src/queue.js

```diff
@@ -27,14 +27,13 @@ class Queue {
   }
 
   sendMessage (message) {
-    if (this.url) {
-      return this.sqs.sendMessage({
-        MessageBody: message,
-        QueueUrl: this.url
-      }).promise()
-    } else {
-      throw new Error('Queue URL has not been set')
-    }
+    return (this.url ? Promise.resolve() : this.getQueueUrl())
```

Comment:
How does this handle the getQueueUrl promise being rejected?

Reply:
It just returns the rejected promise from `getQueueUrl(...)` as the result of `sendMessage(...)`

Reply:
Can add in a catch at the end to reject with a different error if that is better?

Reply:
I'm not sure that behaviour of the method call makes sense that the rejected promise of another method is returned instead of an expected response from sendMessage